src/dune_rules/jsoo/jsoo_rules.ml

```diff
@@ -450,35 +533,41 @@ let setup_separate_compilation_rules sctx components =
            archive "stdlib.cma" :: archive "std_exit.cmo" :: archives
          | _ -> archives
        in
-       Memo.parallel_iter archives ~f:(fun fn ->
-         let build_context = Context.build_context ctx in
-         let name = Path.basename fn in
-         let dir = in_build_dir build_context ~config [ lib_name ] in
-         let in_context =
-           { Js_of_ocaml.In_context.flags = Js_of_ocaml.Flags.standard
-           ; javascript_files = []
-           ; compilation_mode = None
-           ; sourcemap = None
-           }
-         in
-         let src =
-           let src_dir = Lib_info.src_dir info in
-           Path.relative src_dir name
-         in
-         let target = in_build_dir build_context ~config [ lib_name; with_js_ext name ] in
-         build_cm'
-           sctx
-           ~dir
-           ~in_context
-           ~src
-           ~target
-           ~config:(Some (Action_builder.return config))
-           ~sourcemap:Js_of_ocaml.Sourcemap.Inline
-         |> Super_context.add_rule sctx ~dir))
+       Memo.parallel_iter Js_of_ocaml.Mode.all ~f:(fun mode ->
+         Memo.parallel_iter archives ~f:(fun fn ->
+           let build_context = Context.build_context ctx in
+           let name = Path.basename fn in
+           let dir = in_build_dir build_context ~config [ lib_name ] in
+           let in_context =
+             { Js_of_ocaml.In_context.flags = Js_of_ocaml.Flags.standard
```

Comment:
Should this be `Js_of_ocaml.In_context.default` ?

Reply:
This is your code...
But yes, that should work.

Reply:
> This is your code... 

I know